### ISSUE TITLE
For everyone who is struggling with this module

### ISSUE THREAD

**Issue by sliterok** [jfromaniello/passport.socketio#148]
# This module is easily replaced with 6-12 lines of code: 
- ### 4 of these lines could be found in the [official socket.io example](https://github.com/socketio/socket.io/blob/master/examples/passport-example/index.js#L76-L80)
```javascript
const wrap = middleware => (socket, next) => middleware(socket.request, {}, next);

io.use(wrap(sessionMiddleware));
io.use(wrap(passport.initialize()));
io.use(wrap(passport.session()));
```
> Don't forget to replace `sessionMiddleware` with the one that you've passed to your express server!

- ### If you need to ignore users that aren't authorized you could use the middleware just [below the previous example](https://github.com/socketio/socket.io/blob/master/examples/passport-example/index.js#L82-L88)
```javascript
io.use((socket, next) => {
  if (socket.request.user) {
    next();
  } else {
    next(new Error('unauthorized'))
  }
});
```

- ### Finally for the filterSocketsByUser look here:
```javascript
io.filterSocketsByUser = filterFn =>
  Object.values(io.sockets.connected).filter(socket => socket.handshake && filterFn(socket.conn.request.user))
```

- ### Convinience method for mongo(db/ose) users:
```javascript
io.emitToUser = (_id, event, ...args) =>
  io.filterSocketsByUser(user => user._id.equals(_id)).forEach(socket => socket.emit(event, ...args))
```

**Comment by bliteknight** [comment]
Just leaving this here for any future person struggling to get their session into their sockets...and are using namespaces, make sure to add this middleware to your namespace as well i.e.
```
io.of(<your namespace>).use(wrap(sessionMiddleware)); //if using passport, it will be socket.request.session.passport.user
io.of(<your namespace>).use(wrap(passport.initialize()));
io.of(<your namespace>).use(wrap(passport.session())); // this sets it to socket.request.user
```
socket.io does not share middleware with namespaces.

**Comment by itayganor** [comment]
> socket.io does not share middleware with namespaces.

@bliteknight Is this mentioned in their documentation? couldn't find it anywhere.

**Comment by bliteknight** [comment]
> > socket.io does not share middleware with namespaces.
> 
> @bliteknight Is this mentioned in their documentation? couldn't find it anywhere.

I could not find it either. The example provided here:
https://github.com/socketio/socket.io/blob/master/examples/passport-example/index.js

only works on the default namespace, any other one would not have the middleware shared an no access to the passport user. That was when I realized it must not be sharing it and I had to add them to all my namespaces that required the passport user.

Looks like it is also a feature request: https://github.com/socketio/socket.io/issues/3851
